scripts/autotools/libmongoc/CheckCompression.m4

```diff
@@ -4,20 +4,30 @@ PHP_ARG_WITH([mongodb-snappy],
                              [MongoDB: Enable Snappy for compression [default=auto]])],
              [auto],
              [no])
+PHP_MONGODB_VALIDATE_ARG([PHP_MONGODB_SNAPPY], [auto yes no])
 
 PHP_ARG_WITH([mongodb-zlib],
              [whether to enable zlib for compression],
-             [AS_HELP_STRING([--with-mongodb-zlib=@<:@auto/yes/system/bundled/no@:>@],
+             [AS_HELP_STRING([--with-mongodb-zlib=@<:@auto/system/bundled/no@:>@],
                              [MongoDB: Enable zlib for compression [default=auto]])],
              [auto],
              [no])
 
+dnl PHP_ARG_WITH without a value assigns "yes". Treat it like "auto" since we
```

Comment:
> PHP_ARG_WITH without a value assigns "yes"

FYI, this is possible when passing something like `--with-mongodb-zlib=`. It won't be an issue with PECL install since empty input correctly uses the default value from the `<configureoption>` tag.